libsrc/c64/crt0.s

```diff
@@ -39,35 +38,30 @@ L1:     lda     sp,x
 ; Switch off the BASIC ROM.
 
         lda     $01
-        pha                     ; Remember the value
+        sta     mmusave         ; Save the memory configuration
         and     #$F8
         ora     #$06            ; Enable Kernal+I/O, disable BASIC
         sta     $01
 
-; Clear the BSS data.
-
-        jsr     zerobss
-
-; Save some system settings; and, set up the stack.
-
-        pla
-        sta     mmusave         ; Save the memory configuration
-
         tsx
         stx     spsave          ; Save the system stack ptr
 
-        lda     #<(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
-        sta     sp
-        lda     #>(__RAM_START__ + __RAM_SIZE__ + __STACKSIZE__)
-        sta     sp+1            ; Set argument stack ptr
+; Allow some re-entrancy by skipping the next task if it already was done.
+; This often can let us rerun the program without reloading it.
```

Comment:
I don't know if you remember that there was some time ago a discussion about reRUNing CBM programs. I was pretty surprised back then that even Uz considered this desirable - while I personally don't do at all for the well known reason of wrong content in the DATA segment.
I would have easily dropped reRUNabilty when introducing the INIT handling - and several users would have been pretty unhappy. So your approach seems way more diplomatic ;-))